### PR TITLE
[DF] Improvements to CheckDefine, working towards RDF::Redefine

### PR DESCRIPTION
**Pull request by eguiraud**

### CODE REVIEW

File: tree/dataframe/test/dataframe_interface.cxx [hunk 1]
```diff
@@ -280,12 +280,12 @@ TEST(RDataFrameInterface, InvalidDefine)
    try {
       df.Define("1", [] { return true; });
    } catch (const std::runtime_error &e) {
-      EXPECT_STREQ("Cannot define column \"1\": not a valid C++ variable name.", e.what());
+      EXPECT_STREQ("RDataFrame::Define: cannot define column \"1\". Not a valid C++ variable name.", e.what());
```
Comment (stwunsch) [inline tree/dataframe/test/dataframe_interface.cxx:283]: Do we have a similar test for the `Alias` case?

**Review by stwunsch — APPROVED** [review]
Perfect! :) Probably we're missing some test coverage in the `Alias` case, but otherwise I think it's straight forward.

File: tree/dataframe/test/dataframe_interface.cxx (hunk 1 above)
Comment (eguiraud) [reply tree/dataframe/test/dataframe_interface.cxx:283]: around lines 60 in this file we have similar tests for Alias. I'll add one for the case in which the alias name is not a valid C++ variable name, I think that's missing

Comment (eguiraud) [reply tree/dataframe/test/dataframe_interface.cxx:284]: done

Comment (stwunsch) [reply tree/dataframe/test/dataframe_interface.cxx:284]: Go go go! :)

**Pull request merged**